### PR TITLE
Hamza/eng 127 test scenarios

### PR DESCRIPTION
**Pull request by htahir1**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
This PR basically flips the table on testing. It introduces a global fixture that points all tests to a fresh new repository and global dir. It also introduces a `clean_repo` fixture that users can pass in their tests to get a completely new repository and global dir set up from scratch. This is showcased in the integration tests.

### CODE REVIEW

**Review by AlexejPenner — APPROVED** [review]
Very straigthforward solution :)

File: tests/conftest.py [hunk 1]
```diff
@@ -13,34 +13,82 @@
...
 from zenml.pipelines import pipeline
 from zenml.steps import StepContext, step
 
 
-def pytest_sessionstart(session):
-    """Called after the Session object has been created and
-    before performing collection and entering the run test loop.
-    """
+@pytest.fixture(scope="session", autouse=True)
+def base_repo(tmp_path_factory, session_mocker):
```
Comment (schustmi) [inline tests/conftest.py:30]: So this is getting cached and not being run on every test right?

Comment (schustmi) [inline tests/conftest.py:30]: So this is getting cached and not being run on every test right?

File: tests/conftest.py [hunk 2]
```diff
@@ -13,34 +13,82 @@
...
+
+    # clean up
+    shutil.rmtree(tmp_path)
+
+
+@pytest.fixture
+def clean_repo(tmp_path_factory, mocker, base_repo: Repository):
+    """Fixture to get a clean repository for an individual test."""
+    # set env variables
+    os.environ[ENV_ZENML_DEBUG] = "true"
```
Comment (schustmi) [inline tests/conftest.py:67]: Can we get away with removing this duplicate code as this is calling the base fixture anyway?

File: tests/conftest.py (hunk 1 above)
Comment (htahir1) [reply tests/conftest.py:30]: Yes

File: tests/conftest.py (hunk 2 above)
Comment (htahir1) [reply tests/conftest.py]: Indeed. Removed it in my latest commit! Please feel free to check and approve and ill merge :-)

**Review by schustmi — APPROVED** [review]

**Pull request merged**